test/onnx/test_op_consistency.py

```diff
@@ -0,0 +1,462 @@
+    reason: str = "unspecified",
+):
+    """Expects a OpInfo test to fail."""
+    return DecorateMeta(
+        op_name=op_name,
+        variant_name=variant_name,
+        decorator=unittest.expectedFailure,
+        opsets=opsets,
+        device_type=device_type,
+        dtypes=dtypes,
+        reason=reason,
+    )
+
+
+def dont_care(
+    op_name: str,
+    variant_name: str = "",
+    *,
+    opsets: Collection[Union[int, Callable[[int], bool]]] = tuple(),
+    device_type: Optional[str] = None,
+    dtypes: Optional[Collection[torch.dtype]] = None,
+    reason="unspecified",
+):
+    """Skips a test case in OpInfo that we don't care about.
+
+    Likely because ONNX does not support the use case or it is by design.
+    However, if ONNX changes its behavior and start to support the use case, we should
+    update the test to expect the new behavior, leveraging XfailOpset.
+    """
+    return DecorateMeta(
+        op_name=op_name,
+        variant_name=variant_name,
+        decorator=unittest.skip(f"Don't care: {reason}"),
+        opsets=opsets,
+        device_type=device_type,
+        dtypes=dtypes,
+        reason=reason,
+    )
+
+
+def fixme(
+    op_name: str,
+    variant_name: str = "",
+    *,
+    opsets: Collection[Union[int, Callable[[int], bool]]] = tuple(),
+    device_type: Optional[str] = None,
+    dtypes: Optional[Collection[torch.dtype]] = None,
+    reason="unspecified",
+):
+    """Skips a test case in OpInfo. It should be eventually fixed."""
+    return DecorateMeta(
+        op_name=op_name,
+        variant_name=variant_name,
+        decorator=unittest.skip(f"To fix: {reason}"),
+        opsets=opsets,
+        device_type=device_type,
+        dtypes=dtypes,
+        reason=reason,
+    )
+
+
+def skip_ops(
+    all_opinfos: Sequence[opinfo_core.OpInfo],
+    test_class_name: str,
+    base_test_name: str,
+    opset: int,
+    skip_or_xfails: Iterable[DecorateMeta],
+):
+    """Decorates OpInfo tests with decorators based on the skip_or_xfails list."""
+    ops_mapping = {(info.name, info.variant_test_name): info for info in all_opinfos}
+    for decorate_meta in skip_or_xfails:
+        if not decorate_meta.contains_opset(opset):
+            # Skip does not apply to this opset
+            continue
+        opinfo = ops_mapping.get((decorate_meta.op_name, decorate_meta.variant_name))
+        assert (
+            opinfo is not None
+        ), f"Couldn't find OpInfo for {decorate_meta}. Did you need to specify variant_name?"
+        decorators = list(opinfo.decorators)
+        new_decorator = opinfo_core.DecorateInfo(
+            decorate_meta.decorator,
+            test_class_name,
+            base_test_name,
+            device_type=decorate_meta.device_type,
+            dtypes=decorate_meta.dtypes,
+        )
+        decorators.append(new_decorator)
+        opinfo.decorators = tuple(decorators)
+
+    # This decorator doesn't modify fn in any way
+    def wrapped(fn):
```

Comment:
Why make this a wrapper and not just do something like:

```
my_op_db = skip_ops(op_db)
```

?

A more accurate name might be `add_decorate_info`?

Reply:
Good point. I took this from functorch but your suggestion makes more sense here

Reply:
Done